lib/deimos/utils/db_poller/time_based.rb

```diff
@@ -44,20 +44,24 @@ def process_updates
             process_and_touch_info(batch, status)
             time_from = last_updated(batch.last)
           end
-          Deimos.config.logger.info("Poll #{@producer.topic} complete at #{time_to} (#{status.report})")
+          Deimos.config.logger.info("Poll #{log_identifier} complete at #{time_to} (#{status.report})")
         end
 
         # @param time_from [ActiveSupport::TimeWithZone]
         # @param time_to [ActiveSupport::TimeWithZone]
         # @return [ActiveRecord::Relation]
         def fetch_results(time_from, time_to)
-          id = @producer.config[:record_class].primary_key
+          id = if @config.producer_class.present?
+                 @resource_class.config[:record_class].primary_key
+               else
+                 @resource_class.producers.first.config[:record_class].primary_key
+               end
```

Comment:
You should be able to replace this with just

`@resource_class.producer_classes.first.config[:record_class].primary_key`

This should always return the right field.

Reply:
@dorner Would this work considering that the `@resource_class` variable can be set to the constantized producer class as set in the initialize method:
`@resource_class = self.class.producers.any? ? self.class : @config.producer_class.constantize`
and the producer class doesn't have the `producer_classes` method since that method lives in the DbPoller class?

Reply:
Sorry, my bad - should be `self.producer_classes`

Reply:
No worries. Sounds good 👍